sklearn/datasets/_base.py

```diff
@@ -1157,13 +1224,30 @@ def load_boston(*, return_X_y=False):
         .. versionchanged:: 0.20
             Fixed a wrong data point at [445, 0].
 
+    References
+    ----------
+    .. [1] `Racist data destruction? M Carlisle,
+            <https://medium.com/@docintangible/racist-data-destruction-113e3eff54a8>`_
+    .. [2] `Harrison Jr, David, and Daniel L. Rubinfeld.
+           "Hedonic housing prices and the demand for clean air."
+           Journal of environmental economics and management 5.1 (1978): 81-102.
+           <https://www.researchgate.net/publication/4974606_Hedonic_housing_prices_and_the_demand_for_clean_air>`_
+    .. [3] `California housing dataset
+            <https://scikit-learn.org/stable/datasets/real_world.html#california-housing-dataset>`_
+    .. [4] `Ames housing dataset
+            <https://www.openml.org/d/42165>`_
+
     Examples
     --------
     >>> from sklearn.datasets import load_boston
     >>> X, y = load_boston(return_X_y=True)
     >>> print(X.shape)
     (506, 13)
     """
+    # TODO: Implement a module level `__getattr__` function
+    # in`sklearn.datasets` to serve a similar purpose but would raise the
+    # exception at import time instead of call time. See:
+    # https://www.python.org/dev/peps/pep-0562/
```

Comment:
```suggestion
    # TODO: once the deprecation period is over, implement a module level
    # `__getattr__` function in`sklearn.datasets` to raise an exception with
    # an informative error message at import time instead of just removing
    # load_boston. The goal is to avoid having beginners that copy-paste code
    # from numerous books and tutorials that use this dataset loader get 
    # a confusing ImportError when trying to learn scikit-learn.
    # See: https://www.python.org/dev/peps/pep-0562/
```